core/src/test/java/org/apache/servicecomb/core/filter/SimpleRetryFilter.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.servicecomb.core.filter;
+
+import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+
+import org.apache.servicecomb.core.Invocation;
+import org.apache.servicecomb.core.exception.Exceptions;
+import org.apache.servicecomb.swagger.invocation.Response;
+
+@FilterMeta(name = "simple-retry")
+public class SimpleRetryFilter implements Filter {
+  protected int maxRetry = 3;
+
+  public SimpleRetryFilter setMaxRetry(int maxRetry) {
+    this.maxRetry = maxRetry;
+    return this;
+  }
+
+  @Override
+  public CompletableFuture<Response> onFilter(Invocation invocation, FilterNode nextNode) {
+    return new RetrySession(invocation, nextNode).run();
+  }
+
+  protected Throwable unwrapException(Throwable throwable) {
+    return Exceptions.unwrapIncludeInvocationException(throwable);
+  }
+
+  protected boolean isRetryException(Throwable throwable) {
+    return !(throwable instanceof IOException);
```

Comment:
How do you judge which is reatrable？ This is different than current DefaultRetryExtensionsFactory.isRetriableException implementation.